streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -333,22 +333,28 @@ public void onRestoreEnd(final TopicPartition topicPartition, final String store
             offsetsByTopicPartition.put(tp, new AtomicLong());
         }
         consumer.assign(partitionsByTopic.values());
+        final Map<TopicPartition, Long> startOffsets = new HashMap<>();
+        for (final TopicPartition topicPartition : partitionsByTopic.values()) {
+            startOffsets.put(topicPartition, 0L);
+        }
+        consumer.updateBeginningOffsets(startOffsets);
 
         if (globalTopology != null) {
+            final MockConsumer<byte[], byte[]> globalConsumer = new MockConsumer<>(OffsetResetStrategy.NONE);
```

Comment:
We cannot share the consumer any longer, because the global task calls `unassign()` that nukes our setup from above.

Reply:
Not clear why? In MockConsumer we only do the following:

```
public synchronized void unsubscribe() {
        ensureNotClosed();
        committed.clear();
        subscriptions.unsubscribe();
    }
```

And the beginningOffsets map are not nuked.

Reply:
The problem is that the `subscription` is nuked and when we call `position()` the `MockConsumer` checks if the passed in partition is in its subscription and fails before it tries to access the `beginningOffsets` map.

Reply:
Ah okay, got it.